File: Python/dynamic-programming/maximum_subarray.py
from typing import List

def maxSubArray(nums: List[int]) -> int:
    n = len(nums)
    result = nums[0]
    for i in range(1, n):
        if nums[i] < nums[i] + nums[i - 1]:
            nums[i] = nums[i] + nums[i - 1]
        result = max(result, nums[i])

    return result

if __name__ == "__main__":
    print(maxSubArray([-1000]))    

File: Java/src/com/dbc/algorithm/DynamicProgramming/MaximumSubarray.java
package com.dbc.algorithm.DynamicProgramming;

public class MaximumSubarray {
    public int maxSubArray(int[] nums) {
        int result = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] < nums[i] + nums[i - 1]) {
                nums[i] = nums[i] + nums[i - 1];
            }
            result = Math.max(result, nums[i]);
        }

        return result;
    }
}
